scripts/each-service.js

```diff
@@ -0,0 +1,118 @@
+// Helper script that can read the set of services included
+// in the azure SDK dynamically.
+
+var fs = require('fs');
+var path = require('path');
+var util = require('util');
+
+var sdkRoot = path.join(__dirname, '..');
+var servicesRoot = path.join(sdkRoot, 'lib/services');
+
+//
+// Can't use readable-stream, have to only use core modules
+// since this is called by ci before npm install runs.
+//
+
+//
+// Implementation of async versions of fundamental higher-order functions
+// over arrays.
+//
+function reduceAsync(items, asyncIterator, done, initialValue) {
+  if (!initialValue) { initialValue = []; }
+  function doReduce(accumulator, remainingItems, index) {
+    if (remainingItems.length === 0) {
+      return done(null, accumulator);
+    }
+
+    asyncIterator(accumulator, remainingItems[0],
+      function (err, newAccumulator) {
+        if (err) { return done(err); }
+        doReduce(newAccumulator, remainingItems.slice(1), index + 1);
+      }, index, items);
+  }
+  doReduce(initialValue, items, 0);
+}
+
+function filterAsync(items, asyncPredicate, done) {
+  reduceAsync(items,
+    function(acc, item, next) {
+      asyncPredicate(item, function(err, passed) {
+          if (err) { return next(err); }
```

Comment:
I actually did this extra level of indentation deliberately in this case. I was getting lost in all the nested callbacks. Basically, the code is indented as if the preceding "function() {" was on it's own line. I found it much easier to read this way. I can bump the functions down a line to make everything line up if you'd like.